lib/tapioca/active_record_column_type_helper.rb

```diff
@@ -0,0 +1,98 @@
+# typed: strict
+# frozen_string_literal: true
+
+class ActiveRecordColumnTypeHelper
+  extend T::Sig
+
+  sig { params(constant: T.class_of(ActiveRecord::Base)).void }
+  def initialize(constant)
+    @constant = constant
+  end
+
+  sig { params(column_name: String).returns([String, String]) }
+  def type_for(column_name)
+    return ["T.untyped", "T.untyped"] if do_not_generate_strong_types?(@constant)
+
+    column_type = @constant.attribute_types[column_name]
+
+    getter_type =
+      case column_type
+      when defined?(MoneyColumn) && MoneyColumn::ActiveRecordType
+        "::Money"
+      when ActiveRecord::Type::Integer
+        "::Integer"
+      when ActiveRecord::Type::String
+        "::String"
+      when ActiveRecord::Type::Date
+        "::Date"
+      when ActiveRecord::Type::Decimal
+        "::BigDecimal"
+      when ActiveRecord::Type::Float
+        "::Float"
+      when ActiveRecord::Type::Boolean
+        "T::Boolean"
+      when ActiveRecord::Type::DateTime, ActiveRecord::Type::Time
+        "::DateTime"
+      when ActiveRecord::AttributeMethods::TimeZoneConversion::TimeZoneConverter
+        "::ActiveSupport::TimeWithZone"
+      else
+        handle_unknown_type(column_type)
+      end
+
+    column = @constant.columns_hash[column_name]
+    setter_type = getter_type
+
+    if column&.null
+      return ["T.nilable(#{getter_type})", "T.nilable(#{setter_type})"]
+    end
+
+    if column_name == @constant.primary_key ||
+        column_name == "created_at" ||
+        column_name == "updated_at"
+      getter_type = "T.nilable(#{getter_type})"
+    end
+
+    [getter_type, setter_type]
+  end
+
+  private
+
+  sig { params(constant: Module).returns(T::Boolean) }
+  def do_not_generate_strong_types?(constant)
+    Object.const_defined?(:StrongTypeGeneration) &&
+        !(constant.singleton_class < Object.const_get(:StrongTypeGeneration))
+  end
+
+  sig { params(column_type: Object).returns(String) }
+  def handle_unknown_type(column_type)
+    return "T.untyped" unless ActiveModel::Type::Value === column_type
+
+    lookup_return_type_of_method(column_type, :deserialize) ||
```

Comment:
In this case, since we are looking for a specific method from an Active Model value, I think it's better to use the specific arguments.

Generally, hashes as arguments tend to be used for generic option parameters or a set of related parameters, but not for specific ones.

If we define the argument as a hash, any hash will be valid for that argument (even an empty hash), which tends to be too loose for most cases. In this particular method, having the column type and the method name is mandatory, so a hash or an array would be too broad.